zoom_rooms/indico_zoom_rooms/plugin_test.py

```diff
@@ -0,0 +1,841 @@
+
+    create_zoom_meeting(event, 'event').events[0]
+
+    return event
+
+
+@pytest.fixture
+def zr_contrib(
+    zr_event,
+    create_contribution,
+    create_timetable_entry,
+    create_zoom_meeting,
+    zr_powered_room,
+    another_zr_powered_room,
+    mock_tasks,
+):
+    contrib = create_contribution(title='Test Contribution #1', event=zr_event)
+
+    # schedule contribution at 16:15
+    create_timetable_entry(zr_event, contrib, start_dt=datetime(2024, 3, 1, 16, 15, 0, tzinfo=TZ))
+    # create a zoom meeting for the contribution
+    create_zoom_meeting(
+        contrib,
+        'contribution',
+    ).events[0]
+
+    return contrib
+
+
+@pytest.fixture
+def zr_session(zr_event, zr_powered_room, app_context, dummy_room, create_zoom_meeting, mock_tasks):
+    with app_context.test_request_context():
+        session = create_session(zr_event, {'title': 'Test Session #1'})
+
+        block_entry_1 = create_session_block_entry(
+            session, {'duration': timedelta(minutes=30), 'start_dt': datetime(2024, 3, 1, 16, 15, 0, tzinfo=TZ)}
+        )
+        block_entry_2 = create_session_block_entry(
+            session,
+            {
+                'duration': timedelta(minutes=30),
+                'location_data': {'room': dummy_room, 'venue': dummy_room.location, 'inheriting': False},
+                'start_dt': datetime(2024, 3, 1, 17, 15, 0, tzinfo=TZ),
+            },
+        )
+
+    block_1 = block_entry_1.session_block
+    block_2 = block_entry_2.session_block
+
+    contrib_1 = create_contribution(
+        zr_event, {'title': 'Test Contribution #1', 'duration': timedelta(minutes=20)}, session_block=block_1
+    )
+    contrib_2 = create_contribution(
+        zr_event, {'title': 'Test Contribution #2', 'duration': timedelta(minutes=20)}, session_block=block_2
+    )
+
+    # schedule contribution 1 at 16:15
+    schedule_contribution(contrib_1, datetime(2024, 3, 1, 16, 15, 0, tzinfo=TZ), session_block=block_1)
+    # schedule contribution 2 at 17:15
+    schedule_contribution(contrib_2, datetime(2024, 3, 1, 17, 15, 0, tzinfo=TZ), session_block=block_2)
+
+    # create a zoom meeting for contribution 1
+    create_zoom_meeting(
+        contrib_1,
+        'contribution',
+    ).events[0]
+
+    # create a zoom meeting for contribution 2
+    create_zoom_meeting(
+        contrib_2,
+        'contribution',
+    ).events[0]
+
+    # create a zoom meeting for session block 1
+    create_zoom_meeting(
+        block_1,
+        'block',
+    ).events[0]
+    # create a zoom meeting for session block 2
+    create_zoom_meeting(
+        block_2,
+        'block',
+    ).events[0]
+
+    return (
+        session,
+        block_1,
+        block_2,
+        contrib_1,
+        contrib_2,
+    )
+
+
+EVENT_ENTRY_ID = 'zoom_meeting:test-zoom-room-1@example.com@indico:event:1#zmeeting1'
+ENTRY_TITLE = 'Zoom Meeting'
+
+
+def test_event(zr_event, mock_tasks, dummy_room, zr_powered_room):
+    tasks = mock_tasks()
```

Comment:
Is this needed at the beginning of the test?

Reply:
hm... how do I get the tasks iterator otherwise?

Reply:
huh i could've sworn I commented this on some reset call...